concepts/iterators/introduction.md

```diff
@@ -0,0 +1,173 @@
+    type Item;
+    type IntoIter: Iterator;
+    fn into_iter(self) -> Self::IntoIter;
+}
+```
+
+An understanding of `traits` is not necessary for this concept and we will have a whole exercise for you to master `traits`.
+For now, just remember that:
+
+- Any type that implements `IntoIterator` is called an _iterable_ because it can be turned into an iterator.
+- And any type that implements `Iterator` is an iterator, which _produces_ values and can be _iterated over_.
+- You can get an iterator from any type that implements `IntoIterator` by calling its `into_iter` method.
+
+## `into_iter` and for loops
+
+As you can see, any type that implements `IntoIterator` has an `into_iter` method.
+In fact, you're probably familiar with for loops over a vector (`Vec<T>`):
+
+```rust
+// capitals is a `Vec`, which implements `IntoIterator`
+let lor = vec!["Ainur", "Elves", "Men", "Dwarves", "Hobbits", "Orcs", "Dragons"];
+for people in &lor {
+    println!("{}", people);
+}
+```
+
+which are just a shorthand for calling the vector's `into_iter` method to turn the `iterable` into an `iterator`
+and then calls the `next` method on the `iterator` until it produces no more values (i.e. returns `None`):
+
+```rust
+// capitals is a `Vec`, which implements `IntoIterator`
+let mut peoples = (&lor).into_iter();
+
+// capitals_iterator implements `Iterator`, which has the default `next` method
+while let Some(people) = peoples.next() {
+    println!("{}", people);
+}
+```
+
+## `iter` and `iter_mut` methods
+
+Most Rust collection types (such as `Vec`, `HashMap`, `HashSet`) provide `iter` and `iter_mut` methods.
+For an iterator that produces values of type `T`, these two methods perform the following by convention:
+
+- The iterator returned by `iter` will yield immutably borrowed `&T`.
+- The iterator returned by `iter_mut` will yield mutably borrowed `&mut T`.
+
+```rust
+let mut v = vec![1, 2, 3];
+
+// Using `iter` method
+let mut iterator = v.iter();
+assert_eq!(iterator.next(), Some(&1));
+assert_eq!(iterator.next(), Some(&2));
+assert_eq!(iterator.next(), Some(&3));
+assert_eq!(iterator.next(), None);
+
+// Using `iter_mut` method
+let mut muterator = v.iter_mut();
+assert_eq!(muterator.next(), Some(&mut 1));
+assert_eq!(muterator.next(), Some(&mut 2));
+assert_eq!(muterator.next(), Some(&mut 3));
+assert_eq!(muterator.next(), None);
+```
+
+## Difference between `iter`, `iter_mut`, and `into_iter`
+
+Now that we've covered `iter`, `iter_mut`, and `into_iter`, what are the differences?
+Let's amend our previous bullet points:
+
+- The iterator returned by `iter` will yield immutably borrowed `&T`.
+- The iterator returned by `iter_mut` will yield mutably borrowed `&mut T`.
+- The iterator returned by `into_iter` may yield `T`, `&T` or `&mut T`, depending on its context.
+
+Wait a second, what do we mean by _depending on its context_?
+Just what `into_iter` is invoked on.
+Let's clarify using the following bullet points:
```

Comment:
```suggestion
What is the context for an `into_iter` invocation?
Its `self` type.
Recall that there are `IntoIterator` implementations for `Vec`, `&'a Vec`, and `&'a mut Vec`.
Therefore: 
```

Aiming for somewhat more of a reference style; somewhat less of a conversational style.

Reply:
Sure! I was trying conversational because Jeremy mentioned it in last Friday's sync, but for the track it's better to stick to one style.